src/core_functions/scalar/date/date_part.cpp

```diff
@@ -1421,12 +1521,16 @@ struct StructDatePart {
 					ConstantVector::SetNull(*child_entry, false);
 					const auto part_index = size_t(info.part_codes[col]);
 					if (owners[part_index] == col) {
-						part_values[part_index] = ConstantVector::GetData<int64_t>(*child_entry);
+						if (part_index < size_t(BEGIN_DOUBLE)) {
+							part_bigints[part_index - BEGIN_BIGINT] = ConstantVector::GetData<int64_t>(*child_entry);
+						} else {
+							part_doubles[part_index - BEGIN_DOUBLE] = ConstantVector::GetData<double>(*child_entry);
+						}
 					}
 				}
 				auto tdata = ConstantVector::GetData<INPUT_TYPE>(input);
 				if (Value::IsFinite(tdata[0])) {
-					DatePart::StructOperator::Operation(part_values.data(), tdata[0], 0, part_mask);
+					DatePart::StructOperator::Operation(part_bigints, part_doubles, tdata[0], 0, part_mask);
```

Comment:
Is there a reason we use the raw `int **` data of the vector instead of using the vector normally?
Since we have checked indexing on vectors now I would prefer to use it here

Reply:
Old code - good point!